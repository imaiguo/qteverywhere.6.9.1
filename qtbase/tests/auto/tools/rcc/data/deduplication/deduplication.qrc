<!DOCTYPE RCC><RCC version="1.0">
<qresource>
   <file>files/a.txt</file>
   <file>files/b.txt</file>
   <file alias="files/alias_of_b.txt">files/b.txt</file>
   <file>files/c_with_a_content.txt</file>
   <file alias="files/alias_of_b_compress9.txt" compress="9">files/b.txt</file>
   <file alias="files/alias_of_b_compress9_dupe.txt" compress="9">files/b.txt</file>
</qresource>
</RCC>
