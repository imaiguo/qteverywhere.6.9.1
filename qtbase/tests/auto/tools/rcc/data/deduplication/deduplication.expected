/****************************************************************************
** Resource object code
**
IGNORE:** Created by: The Resource Compiler for Qt version 6.9.0
**
** WARNING! All changes made in this file will be lost!
*****************************************************************************/

#ifdef _MSC_VER
// disable informational message "function ... selected for automatic inline expansion"
#pragma warning (disable: 4711)
#endif

static const unsigned char qt_resource_data[] = {
  // b.txt
  0x0,0x0,0x0,0xb,
  0x62,
  0x20,0x74,0x65,0x73,0x74,0x20,0x66,0x69,0x6c,0x65,
    // c_with_a_content.txt
  0x0,0x0,0x0,0xb,
  0x61,
  0x20,0x74,0x65,0x73,0x74,0x20,0x66,0x69,0x6c,0x65,
    // b.txt
  0x0,0x0,0x0,0xb,
  0x62,
  0x20,0x74,0x65,0x73,0x74,0x20,0x66,0x69,0x6c,0x65,
  
};

static const unsigned char qt_resource_name[] = {
  // files
  0x0,0x5,
  0x0,0x6d,0x2,0xc3,
  0x0,0x66,
  0x0,0x69,0x0,0x6c,0x0,0x65,0x0,0x73,
    // b.txt
  0x0,0x5,
  0x0,0x65,0x5b,0xf4,
  0x0,0x62,
  0x0,0x2e,0x0,0x74,0x0,0x78,0x0,0x74,
    // c_with_a_content.txt
  0x0,0x14,
  0x1,0x61,0x1d,0x34,
  0x0,0x63,
  0x0,0x5f,0x0,0x77,0x0,0x69,0x0,0x74,0x0,0x68,0x0,0x5f,0x0,0x61,0x0,0x5f,0x0,0x63,0x0,0x6f,0x0,0x6e,0x0,0x74,0x0,0x65,0x0,0x6e,0x0,0x74,0x0,0x2e,
  0x0,0x74,0x0,0x78,0x0,0x74,
    // a.txt
  0x0,0x5,
  0x0,0x64,0x5b,0xf4,
  0x0,0x61,
  0x0,0x2e,0x0,0x74,0x0,0x78,0x0,0x74,
    // alias_of_b_compress9.txt
  0x0,0x18,
  0xb,0x26,0xf,0xb4,
  0x0,0x61,
  0x0,0x6c,0x0,0x69,0x0,0x61,0x0,0x73,0x0,0x5f,0x0,0x6f,0x0,0x66,0x0,0x5f,0x0,0x62,0x0,0x5f,0x0,0x63,0x0,0x6f,0x0,0x6d,0x0,0x70,0x0,0x72,0x0,0x65,
  0x0,0x73,0x0,0x73,0x0,0x39,0x0,0x2e,0x0,0x74,0x0,0x78,0x0,0x74,
    // alias_of_b.txt
  0x0,0xe,
  0x1,0xa4,0x6d,0x34,
  0x0,0x61,
  0x0,0x6c,0x0,0x69,0x0,0x61,0x0,0x73,0x0,0x5f,0x0,0x6f,0x0,0x66,0x0,0x5f,0x0,0x62,0x0,0x2e,0x0,0x74,0x0,0x78,0x0,0x74,
    // alias_of_b_compress9_dupe.txt
  0x0,0x1d,
  0x9,0x4,0x7a,0x14,
  0x0,0x61,
  0x0,0x6c,0x0,0x69,0x0,0x61,0x0,0x73,0x0,0x5f,0x0,0x6f,0x0,0x66,0x0,0x5f,0x0,0x62,0x0,0x5f,0x0,0x63,0x0,0x6f,0x0,0x6d,0x0,0x70,0x0,0x72,0x0,0x65,
  0x0,0x73,0x0,0x73,0x0,0x39,0x0,0x5f,0x0,0x64,0x0,0x75,0x0,0x70,0x0,0x65,0x0,0x2e,0x0,0x74,0x0,0x78,0x0,0x74,
  
};

static const unsigned char qt_resource_struct[] = {
  // :
  0x0,0x0,0x0,0x0,0x0,0x2,0x0,0x0,0x0,0x1,0x0,0x0,0x0,0x1,
0x0,0x0,0x0,0x0,0x0,0x0,0x0,0x0,
  // :/files
  0x0,0x0,0x0,0x0,0x0,0x2,0x0,0x0,0x0,0x6,0x0,0x0,0x0,0x2,
0x0,0x0,0x0,0x0,0x0,0x0,0x0,0x0,
  // :/files/a.txt
  0x0,0x0,0x0,0x4e,0x0,0x0,0x0,0x0,0x0,0x1,0x0,0x0,0x0,0xf,
TIMESTAMP:files/a.txt
  // :/files/b.txt
  0x0,0x0,0x0,0x10,0x0,0x0,0x0,0x0,0x0,0x1,0x0,0x0,0x0,0x0,
TIMESTAMP:files/b.txt
  // :/files/c_with_a_content.txt
  0x0,0x0,0x0,0x20,0x0,0x0,0x0,0x0,0x0,0x1,0x0,0x0,0x0,0xf,
TIMESTAMP:files/c_with_a_content.txt
  // :/files/alias_of_b.txt
  0x0,0x0,0x0,0x94,0x0,0x0,0x0,0x0,0x0,0x1,0x0,0x0,0x0,0x0,
TIMESTAMP:files/b.txt
  // :/files/alias_of_b_compress9_dupe.txt
  0x0,0x0,0x0,0xb6,0x0,0x0,0x0,0x0,0x0,0x1,0x0,0x0,0x0,0x1e,
TIMESTAMP:files/b.txt
  // :/files/alias_of_b_compress9.txt
  0x0,0x0,0x0,0x5e,0x0,0x0,0x0,0x0,0x0,0x1,0x0,0x0,0x0,0x1e,
TIMESTAMP:files/b.txt

};

#ifdef QT_NAMESPACE
#  define QT_RCC_PREPEND_NAMESPACE(name) ::QT_NAMESPACE::name
#  define QT_RCC_MANGLE_NAMESPACE0(x) x
#  define QT_RCC_MANGLE_NAMESPACE1(a, b) a##_##b
#  define QT_RCC_MANGLE_NAMESPACE2(a, b) QT_RCC_MANGLE_NAMESPACE1(a,b)
#  define QT_RCC_MANGLE_NAMESPACE(name) QT_RCC_MANGLE_NAMESPACE2( \
        QT_RCC_MANGLE_NAMESPACE0(name), QT_RCC_MANGLE_NAMESPACE0(QT_NAMESPACE))
#else
#   define QT_RCC_PREPEND_NAMESPACE(name) name
#   define QT_RCC_MANGLE_NAMESPACE(name) name
#endif

#if defined(QT_INLINE_NAMESPACE)
inline namespace QT_NAMESPACE {
#elif defined(QT_NAMESPACE)
namespace QT_NAMESPACE {
#endif

bool qRegisterResourceData(int, const unsigned char *, const unsigned char *, const unsigned char *);
bool qUnregisterResourceData(int, const unsigned char *, const unsigned char *, const unsigned char *);

#ifdef QT_NAMESPACE
}
#endif

int QT_RCC_MANGLE_NAMESPACE(qInitResources)();
int QT_RCC_MANGLE_NAMESPACE(qInitResources)()
{
    int version = 3;
    QT_RCC_PREPEND_NAMESPACE(qRegisterResourceData)
        (version, qt_resource_struct, qt_resource_name, qt_resource_data);
    return 1;
}

int QT_RCC_MANGLE_NAMESPACE(qCleanupResources)();
int QT_RCC_MANGLE_NAMESPACE(qCleanupResources)()
{
    int version = 3;
    QT_RCC_PREPEND_NAMESPACE(qUnregisterResourceData)
       (version, qt_resource_struct, qt_resource_name, qt_resource_data);
    return 1;
}

#ifdef __clang__
#   pragma clang diagnostic push
#   pragma clang diagnostic ignored "-Wexit-time-destructors"
#endif

namespace {
   struct initializer {
       initializer() { QT_RCC_MANGLE_NAMESPACE(qInitResources)(); }
       ~initializer() { QT_RCC_MANGLE_NAMESPACE(qCleanupResources)(); }
   } dummy;
}

#ifdef __clang__
#   pragma clang diagnostic pop
#endif
